### DISCUSSION TITLE
Resize according to orientation

### DISCUSSION THREAD

**Discussion by tomaszroot** [ImageMagick/ImageMagick#5772] [Help] [answered]
I need to resize images in bulk. For landscapes let it be 1920 px for the long edge and for portrait let it be 1280 px for the short edge.
Currently I have:
`for file in *.jpg; do convert -auto-orient -resize 1920 $file $file; done`
In my case, it only changes the landscape files.

**Comment by snibgo** [comment]
I'm not sure exactly what you want. I suggest you write a script that first calculates if your image is landscape or portrait format, then executes one IM command for portraits and a different IM command for landscapes.

When that does what you want, perhaps it can be rolled up into a single IM command.

**Comment by fmw42** [comment, marked as answer]
Does this do what you want?

`for file in *.jpg; do convert -auto-orient -resize 1920x1280 $file $file; done`

If on IM 7, you can determine if portrait or landscape as an inline computation for -resize. 

Please identify your IM version and platform/OS when asking questions.

**Reply by tomaszroot** [reply]
Yes exactly, this is what I was looking for. Thank you.

**Comment by jungle-boogie** [comment]
I have a similar question, but I want my aspect ratio to be ignored so that I can make 5x7 prints.

This works for landscape images:
`convert input.jpg -auto-orient -resize 1260x900!  output.jpg`

This work for portrait images:
`convert input.jpg -auto-orient -resize 900x1200!  output.jpg`

But how I can i have one command work for the two different types of images to resize them to either 5x7 or 7x5?

From what I can determine, the `-auto-orient` has no effect; maybe that doesn't work when the aspect ratio is ignored?

**Comment by fmw42** [comment]
If you use Imagemagick 7, you can do that in one command line (rather than make variables first) as follows.  It tests if w>=h and does one thing otherwise (h>w) it does the other.

`magick input -auto-orient -set option:wd "%[fx:(w>=h)?1260:900]!" -set option:ht "%[fx:(w>=h)?900:1260]!" -resize "%[wd]x%[ht]" output.jpg
`
